### ISSUE TITLE
How to replace OrchardCMS Field default Template in the custom module, please?

### ISSUE THREAD

**Issue by effapp** [OrchardCMS/OrchardCore#14631]
How to replace OrchardCMS Field default Template in the custom module, please?

**Comment by MikeAlhayek** [comment]
I think you have to override it from a theme not a module. if you want a custom module, then maybe you should use a different editor of that field that your module exposes.

**Comment by effapp** [comment]
If I override it from a theme not a module, Template will be invalid after Customer change the theme .



> I think you have to override it from a theme not a module. if you want a custom module, then maybe you should use a different editor of that field that your module exposes.

**Comment by MikeAlhayek** [comment]
Yes that is correct. You could try using the Admin Template feature and create a template using the UI. Otherwise, you'll have to create a customer editor for your field that would be available when your feature is enabled.

Feature, cannot override template that were composed by other features. At least, I am not aware of a way to do it. You could also override it from the web project which will override everything (themes and modules)

**Comment by jtkech** [comment]
You can override a template from a module.

If you override a template file with the exact same name, your module/feature should depend on the module that defines the original file, the dependency can be defined in your module Manifest file (there many examples in the source), so that you have an higher precedence. You can use the exact same file name or a more specific one depending on the Type/Part/Field names (refer to the doc for the file names, here too there are many example in the source).

Unless if the current Theme already has the same file, the current Theme has an higher precedence. In that case your module can still define more or less specific template files but with different names.

The Template feature has the highest precedence so you can always override a template from the admin UI, here the name should follow the template names (different than file names) conventions, refer to the doc, from memory a `.` becomes a `_` and a `-` becomes a double underscore `__`, and for example the display type, which is at the end for a file name, is at the 2nd position (after the shape type) for a template name.

**Comment by effapp** [comment]
Overriding Field Template in views folder of web is valid, but Overriding Field Template in views folder of module is invalid.

**Comment by MikeAlhayek** [comment]
@effapp I never tried @jtkech. But if he said it can be done, then it can :)

What he is saying that the dependency here matter. Ensure your module add reference dependency to OC.ContentFields project. Then add the view to your module which should force the view locator to find the view defined in your module.

Another way you could try is by overriding the `Order` property in your Startup class. Try setting it to 1 and see if that works. If that does not work, try -1 and see.

Anyway, I am converting this to a discussion.
